feat: proguard processor

martin308 opened the pull request:
Adds the proguard processor. A lot of churn in the go.mod/sum files due to workspace shenanigans

martin308 committed:
Basic proguard symbolication

martin308 committed:
go work sync

wolfgangcodes reviewed (approved):
I don't have the deepest context here, so maybe @MustafaHaddara can give an add'l 👀  here?

MustafaHaddara reviewed (approved):
LGTM 👍

martin308 committed:
Only preserve stack trace if needed

martin308 committed:
Update key name

martin308 committed:
Add dependabot config

MustafaHaddara commented on proguardprocessor/factory.go:
```diff
@@ -0,0 +1,83 @@
+		OriginalMethodsAttributeKey: "exception.structured_stacktrace.methods.original",
+		OriginalLinesAttributeKey:   "exception.structured_stacktrace.lines.original",
+		// TODO: determine if this is a good default
+		ProguardUUIDAttributeKey: "exception.structured_stacktrace.proguard_uuid",
```
well I used `app.debug.build_uuid` in https://github.com/honeycombio/opentelemetry-collector-symbolicator/pull/73

martin308 committed:
Update Proguard UUID attribute key in config

Changed the ProguardUUIDAttributeKey from 'build.proguard.uuid' to 'app.debug.proguard_uuid' in the default configuration to align with updated attribute naming conventions.
